src/Home.tsx

```diff
@@ -0,0 +1,264 @@
+					return
+				}
+
+				const uint8Data = new Uint8Array(indexData);
+				const objectsData: string[] = [];
+				for (let i = 0; i < uint8Data.length; i += 32) {
+					const chunk = uint8Data.slice(i, i + 32);
+					const encoded = base58.encode(chunk);
+					objectsData.push(encoded);
+				}
+
+				for (let i = 0; i <= objectsData.length; i += 1) {
+					if (blockCount <= (indexFile * 128000 + i)) {
+						await writableStream.close();
+						return
+					}
+
+					const objectData: Uint8Array | string = await fetchBlock(currentNet, objectsData[i]);
+					if (typeof objectData === 'string') {
+						onModal('failed', objectData);
+						await writableStream.close();
+						return
+					}
+
+					const blockSize = new Uint32Array([objectData.byteLength]);
+					await writableStream.write(blockSize.buffer);
+					await writableStream.write(new Uint8Array(objectData));
+					setCurrentDownloadedBlock((indexFile * 128000 + i) - formData.spanStart + 1);
+				}
+			}
+
+			await writableStream.close();
+		} catch (error: any) {
+			if (error.message.indexOf('showSaveFilePicker is not a function') !== -1) {
+				onModal('failed', 'Current browser does not support the functionality of the site. For best use, use Chrome 86+ (recommended), Edge 86+ or Opera 72+.');
+			} else {
+				onModal('failed', error.message || 'Error occurred during block fetching.');
+			}
+		} finally {
+			setLoading(false);
+		}
+  };
+
+	const fetchIndexFile = async (currentNet: NetItem, indexNumber: number): Promise<Uint8Array | string> => {
+		try {
+			const searchResponse: any = await api('POST', `/objects/${currentNet.containerId}/search?walletConnect=false&offset=0&limit=1`, {
+				filters: [{
+					"key": "Index",
+					"match": "MatchStringEqual",
+					"value": indexNumber.toString(),
+				}],
+			});
+
+			const objectId = searchResponse.objects[0]?.address.objectId;
+			if (!objectId) {
+				return `Error occurred during index fetching #${indexNumber}`;
+			}
+
+			const indexResponse = await api('GET', `/objects/${currentNet.containerId}/by_id/${objectId}?walletConnect=false`);
+			return indexResponse as Uint8Array;
+		} catch (err: any) {
+			return `Error occurred during index fetching #${indexNumber}: ${err.message}`;
+		}
+	};
+
+	const fetchBlock = async (currentNet: NetItem, objectId: string): Promise<Uint8Array | string> => {
+		try {
+			const blockResponse = await api('GET', `/objects/${currentNet.containerId}/by_id/${objectId}?walletConnect=false`);
+			return blockResponse as Uint8Array;
+		} catch (err: any) {
+			return `Error occurred during object fetching ${objectId}: ${err.message}`;
+		}
+	};
+
+  return (
+		<Container>
+			<Section>
+				<Tile kind="ancestor">
+					<Tile kind="parent">
+						<Tile
+							kind="child"
+							renderAs={Notification}
+							color={"gray"}
+						>
+							<Heading weight="semibold" subtitle style={{ textAlign: 'center' }}>Archive.NeoFS – Download Blockchain Data Snapshot</Heading>
```

Comment:
It's not a blockchain data snapshot, this can be understood as "state snapshot". It is an offline synchronization package, block dump.